lms/services/organization.py

```diff
@@ -58,6 +65,12 @@ def get_by_public_id(self, public_id: str) -> list | None:
 
         :param public_id: Fully qualified public id
         """
+        parts = public_id.split(".")
+
+        if not len(parts) == 4:
```

Comment:
Ok, so `OrganizationService.get_by_public_id` doesn't _have_ to split the `public_id` string (it just needs to compare the string directly to the prefixed strings in the DB), but it's just splitting it so that it can raise this validation error

Reply:
You could easily add this validation to the model as well: https://docs.sqlalchemy.org/en/20/orm/mapped_attributes.html#simple-validators